pandas/tests/frame/test_constructors.py

```diff
@@ -953,6 +947,16 @@ def __len__(self, n):
                             array.array('i', range(10))])
         tm.assert_frame_equal(result, expected, check_dtype=False)
 
+    def test_constructor_iterable(self):
+        class Iter():
+            def __iter__(self):
+                for i in range(10):
+                    yield [1, 2, 3]
+
+        expected = DataFrame([[1, 2, 3]] * 10)
+        result = DataFrame(Iter())
+        tm.assert_frame_equal(result, expected)
+
```

Comment:
Reference the issue number in a comment below the functional definition.  Applies to all newly-added tests.